tla-bmcmt/src/main/scala/at/forsyte/apalache/tla/bmcmt/rules/SetFilterRule.scala

```diff
@@ -58,11 +60,13 @@ class SetFilterRule(rewriter: SymbStateRewriter) extends RewritingRule {
 
         // require each cell to satisfy the predicate
         def addCellCons(cell: ArenaCell, pred: TlaEx): Unit = {
-          val inNewSet = OperEx(TlaSetOper.in, cell.toNameEx, newSetCell.toNameEx)
-          val inOldSet = OperEx(TlaSetOper.in, cell.toNameEx, setCell.toNameEx)
-          val inOldSetAndPred = OperEx(TlaBoolOper.and, pred, inOldSet)
-          val ifAndOnlyIf = OperEx(TlaOper.eq, inNewSet, inOldSetAndPred)
-          rewriter.solverContext.assertGroundExpr(ifAndOnlyIf)
+          val inNewSet = tla.apalacheStoreInSet(cell.toNameEx, newSetCell.toNameEx)
+          val notInNewSet =
+            tla.apalacheUnchangedSet(cell.toNameEx, newSetCell.toNameEx) // This ensures that cell is not in newSetCell
```

Comment:
I still find the name `apalacheUnchangedSet` confusing. It enforces the semantics of the array encoding. In the case of oopsla19 encoding, it is `notin(e, S)`, which is not about the set being unchanged. Shall we call it `apalacheStoreNotInSet`? This would emphasize that this enforces the element to be outside of the set. It just happens that you enforce this constraint as equality in SMT by knowing that a new set is empty.

Reply:
Sure, I was also not happy with the current name, but could not think of a good alternative. Thanks for the suggestion!